phoenix-core-client/src/main/java/org/apache/phoenix/util/ByteUtil.java

```diff
@@ -469,6 +474,105 @@ public static byte[] previousKey(byte[] key) {
         return previousKey;
     }
 
+    public static byte[] getLargestPossibleRowKeyInRange(byte[] startKey, byte[] endKey) {
```

Comment:
This needs documentation comments to describe what it does. It also assumes that startKey and endKey are not null and so it will be good to check them. It looks likes the range here is [startKey, endKey), i.e., endKey is not inclusive. Let's state it in the comments.

Reply:
Done